lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -481,8 +486,10 @@ def return_connection(url)
     end
 
     def get_es_version(url)
-      request = perform_request_to_url(url, :get, ROOT_URI_PATH)
-      LogStash::Json.load(request.body)["version"]["number"]
+      response = perform_request_to_url(url, :get, ROOT_URI_PATH)
+      return nil unless (200..299).cover?(response.code)
+
+      LogStash::Json.load(response.body).dig("version","number")
```

Comment:
:thinking: theoretically the plugin does not limit itself being installed on LS 5.6 (< 6.0) ...
should we do so, since we're explicitly using a Ruby 2.3 (`Hash#dig`) feature here?

Reply:
Good catch. I would love to start using modern ruby features, but the benefit here isn't worth a major version bump of the plugin.